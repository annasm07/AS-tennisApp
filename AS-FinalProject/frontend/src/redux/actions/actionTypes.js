export default {
  LOG_IN: 'LOG_IN',
  LOG_IN_ERROR: 'LOG_IN_ERROR',
  SIGN_UP: 'LOG_IN',
  SIGN_UP_ERROR: 'LOG_IN_ERROR',
  GET_PLAYER: 'GET_PLAYER',
  GET_PLAYER_ERROR: 'GET_PLAYER_ERROR',
  GET_PLAYERS: 'GET_PLAYERS',
  GET_PLAYERS_ERROR: 'GET_PLAYERS_ERROR',
  CREATE_MATCH: 'CREATE_MATCH',
  CREATE_MATCH_ERROR: 'CREATE_MATCH_ERROR',
  UPDATE_POINTS: 'UPDATE_POINTS',
  UPDATE_POINTS_ERROR: 'UPDATE_POINTS_ERROR',
  UPDATE_GAMES: 'UPDATE_GAMES',
  UPDATE_GAMES_ERROR: 'UPDATE_GAMES_ERROR',
  END_GAME: 'END_GAME',
  END_GAME_ERROR: 'END_GAME_ERROR',
  END_SET: 'END_SET',
  END_SET_ERROR: 'END_SET_ERROR',
  END_MATCH: 'END_MATCH',
  END_MATCH_ERROR: 'END_MATCH_ERROR',
};
