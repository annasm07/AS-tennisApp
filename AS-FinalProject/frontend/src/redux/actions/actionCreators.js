import axios from 'axios';
import actionTypes from './actionTypes';

const LOG_IN_URL = 'http://localhost:1616/api/auth/login';
const SIGN_UP_URL = 'http://localhost:1616/api/auth/signup';
const GET_PLAYER_URL = 'http://localhost:1616/api/home';
const GET_ALL_PLAYERS = 'http://localhost:1616/api/players';
const CREATE_MATCH_URL = 'http://localhost:1616/api/match';

export function logIn(email, password) {
  return async dispatch => {
    try {
      const {data} = await axios.post(LOG_IN_URL, {email, password});
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.LOG_IN_ERROR,
      });
    }
  };
}

export function signUp({name, email, password, player, playerName}) {
  return async dispatch => {
    try {
      await axios.post(SIGN_UP_URL, {
        name,
        email,
        password,
        player,
        playerName,
      });
      dispatch({
        // type: actionTypes.SIGN_UP,
        // user: data.user,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.SIGN_UP_ERROR,
      });
    }
  };
}

export function getPlayerInfo(token, playerId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${GET_PLAYER_URL}/${playerId}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYER,
        player: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYER_ERROR,
      });
    }
  };
}
export function getAllPlayers(token) {
  return async dispatch => {
    try {
      const {data} = await axios(GET_ALL_PLAYERS, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYERS,
        players: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYERS_ERROR,
      });
    }
  };
}

export function newMatch(token, p1Name, p2Name) {
  return async dispatch => {
    try {
      const {data} = await axios.post(
        CREATE_MATCH_URL,
        {p1Name, p2Name},
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      );
      dispatch({
        type: actionTypes.CREATE_MATCH,
        currentMatch: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.CREATE_MATCH_ERROR,
      });
    }
  };
}
