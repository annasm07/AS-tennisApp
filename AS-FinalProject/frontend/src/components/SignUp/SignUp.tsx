import React, {useState, useEffect} from 'react';
import {
  Pressable,
  SafeAreaView,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';
import {
  hasNameError,
  hasEmailError,
  hasPasswordError,
} from '../../utils/validation';
import {connect} from 'react-redux';
import globalStyles from '../../theme/globalThemes';
import {signUp} from '../../redux/actions/actionCreators';

const SignUp = ({navigation, dispatch}: any) => {
  const [name, onChangeName] = useState('');
  const [email, onChangeEmail] = useState('');
  const [password, onChangePassword] = useState('');
  const [playerName, onChangePlayerName] = useState('');
  const [isPlayer, setIsPlayer] = useState(true);
  const [registerButttonDisabled, setRegisterButttonDisabled] = useState(true);

  const [nameError, setError] = useState(false);
  const [emailError, setEmailError] = useState(false);
  const [pdwError, setPdwError] = useState(false);

  function handleSignUp() {
    const NewUser = {
      name,
      email,
      password,
      player: isPlayer,
      playerName: playerName || name,
    };
    dispatch(signUp(NewUser));
    navigation.navigate('LogInPage');
  }
  useEffect(() => {
    if (name && email && password && !emailError && !pdwError) {
      setRegisterButttonDisabled(false);
    } else {
      setRegisterButttonDisabled(true);
    }
  }, [name, email, password, emailError, pdwError]);

  return (
    <SafeAreaView style={styles.body}>
      <Text style={styles.title}>Sign Up</Text>
      <TextInput
        style={styles.input}
        onChangeText={onChangeName}
        onEndEditing={() => setError(() => hasNameError(name))}
        placeholder="Full Name"
        value={name}
        autoCapitalize="words"
      />
      {nameError && (
        <View>
          <Text style={styles.messageError}>
            Please, add a name of at least two letters
          </Text>
        </View>
      )}
      <TextInput
        style={styles.input}
        onEndEditing={() => setEmailError(() => hasEmailError(email))}
        onChangeText={onChangeEmail}
        placeholder="Email"
        value={email}
        autoCapitalize="none"
      />
      {emailError && (
        <View>
          <Text style={styles.messageError}>
            E-mail format accepted is: example@example.com
          </Text>
        </View>
      )}
      <TextInput
        style={styles.input}
        onChangeText={onChangePassword}
        onEndEditing={() => setPdwError(() => hasPasswordError(password))}
        value={password}
        secureTextEntry={true}
        placeholder="Password"
        autoCapitalize="none"
      />
      {pdwError && (
        <View>
          <Text style={styles.messageError}>
            Your password has to be at least 5 characters long,
          </Text>
        </View>
      )}
      <View style={styles.label}>
        <Text style={globalStyles.grayText}>Are you a Player or Coach?</Text>
      </View>
      {isPlayer ? (
        <View style={styles.userType}>
          <Pressable
            style={globalStyles.buttonActive}
            onPress={() => setIsPlayer(true)}>
            <Text>Player</Text>
          </Pressable>
          <Pressable
            style={globalStyles.buttonDisabled}
            onPress={() => setIsPlayer(false)}>
            <Text style={globalStyles.grayText}>Coach</Text>
          </Pressable>
        </View>
      ) : (
        <>
          <View style={styles.userType}>
            <Pressable
              style={globalStyles.buttonDisabled}
              onPress={() => setIsPlayer(true)}>
              <Text style={globalStyles.grayText}>Player</Text>
            </Pressable>
            <Pressable
              style={globalStyles.buttonActive}
              onPress={() => setIsPlayer(false)}>
              <Text>Coach</Text>
            </Pressable>
          </View>
          <TextInput
            style={styles.input}
            onChangeText={onChangePlayerName}
            value={playerName}
            placeholder="Player Name"
            autoCapitalize="words"
          />
        </>
      )}
      <TouchableOpacity
        disabled={registerButttonDisabled}
        style={
          registerButttonDisabled ? styles.disabled : globalStyles.buttonYellow
        }
        onPress={handleSignUp}>
        <Text style={registerButttonDisabled && globalStyles.grayText}>
          Sign Up
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={globalStyles.buttonGray}
        onPress={() => navigation.navigate('LogInPage')}>
        <Text>Log in</Text>
      </TouchableOpacity>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  body: {
    top: '20%',
  },
  title: {
    textAlign: 'center',
    fontSize: 24,
    padding: 30,
  },
  userType: {
    marginTop: -20,
    flexDirection: 'row',
    alignSelf: 'center',
  },
  label: {
    marginTop: 14,
    backgroundColor: '#E5E5E5',
    borderRadius: 10,
    fontSize: 14,
    paddingHorizontal: 8,
    paddingVertical: 4,
    marginHorizontal: 30,
    marginBottom: 15,
    flexDirection: 'row',
    justifyContent: 'space-between',
    textDecorationColor: '#7A7A7A',
  },
  input: {
    height: 40,
    margin: 8,
    borderWidth: 1,
    paddingLeft: 20,
    borderRadius: 20,
    fontSize: 16,
    marginHorizontal: 40,
    borderColor: '#7A7A7A',
  },
  disabled: {
    alignItems: 'center',
    alignSelf: 'center',
    backgroundColor: '#F5BF00',
    opacity: 0.5,
    borderRadius: 20,
    borderColor: '#000000',
    borderWidth: 1,
    fontSize: 50,
    marginTop: 30,
    paddingHorizontal: 12,
    paddingVertical: 10,
    width: 120,
  },
  messageError: {
    color: 'red',
    fontSize: 12,
    marginLeft: 50,
  },
});

function mapStateToProps({user}: any) {
  return {
    user,
  };
}

export default connect(mapStateToProps)(SignUp);
