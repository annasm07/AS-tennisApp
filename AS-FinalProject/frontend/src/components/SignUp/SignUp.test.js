import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/actionCreators';
import {Provider} from 'react-redux';
import SignUp from './SignUp';
import {
  hasNameError,
  hasEmailError,
  hasPasswordError,
} from '../../utils/validation';

jest.mock('../../redux/actions/actionCreators');
jest.mock('../../utils/validation');

describe('When invoked a SignIn component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    tokens: ['abc', 'cde'],
  });

  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'signUp').mockReturnValueOnce({type: ''});
    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render SignUp component', () => {
    const signUp = render(
      <Provider store={store}>
        <SignUp navigation={navigation} />
      </Provider>,
    );
    expect(signUp).toMatchSnapshot();
  });

  test('Then hasNameError endEdit is invoked', () => {
    const {getByPlaceholderText} = render(
      <Provider store={store}>
        <SignUp navigation={navigation} />
      </Provider>,
    );
    const imageName = getByPlaceholderText('Full Name');
    fireEvent(imageName, 'onEndEditing', hasNameError);
    expect(hasNameError).toHaveBeenCalled();
  });

  test('Then hasEmailError endEdit is invoked', () => {
    const {getByPlaceholderText} = render(
      <Provider store={store}>
        <SignUp navigation={navigation} />
      </Provider>,
    );
    const imageEmail = getByPlaceholderText('Email');
    fireEvent(imageEmail, 'onEndEditing', hasEmailError);
    expect(hasEmailError).toHaveBeenCalled();
  });

  test('Then hasPasswordError endEdit is invoked', () => {
    const {getByPlaceholderText} = render(
      <Provider store={store}>
        <SignUp navigation={navigation} />
      </Provider>,
    );
    const imagePsswd = getByPlaceholderText('Password');
    fireEvent(imagePsswd, 'onEndEditing', hasPasswordError);
    expect(hasPasswordError).toHaveBeenCalled();
  });

  test('When ChoosePlayer is pressed, call set is player fn', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <SignUp navigation={navigation} />
      </Provider>,
    );
    const ChoosePlayerButton = getByTestId('ChoosePlayerButton1');
    let setIsPlayer = jest.fn();
    fireEvent.press(ChoosePlayerButton, setIsPlayer);
    expect(setIsPlayer).not.toHaveBeenCalled();
  });

  test('When ChooseCoach is pressed, call set is player fn', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <SignUp navigation={navigation} />
      </Provider>,
    );
    const ChooseCoachButton = getByTestId('ChooseCoachButton1');
    let setIsPlayer = jest.fn();
    fireEvent.press(ChooseCoachButton, setIsPlayer);
    expect(setIsPlayer).not.toHaveBeenCalled();
  });
});
