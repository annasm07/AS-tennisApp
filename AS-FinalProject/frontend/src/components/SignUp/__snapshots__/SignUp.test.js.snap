// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a SignIn component Should render SignUp component 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "top": "20%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "padding": 30,
        "textAlign": "center",
      }
    }
  >
    Sign Up
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="words"
    onChangeText={[Function]}
    onEndEditing={[Function]}
    placeholder="Full Name"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#7A7A7A",
        "borderRadius": 20,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 40,
        "margin": 8,
        "marginHorizontal": 40,
        "paddingLeft": 20,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    onEndEditing={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#7A7A7A",
        "borderRadius": 20,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 40,
        "margin": 8,
        "marginHorizontal": 40,
        "paddingLeft": 20,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    onEndEditing={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "borderColor": "#7A7A7A",
        "borderRadius": 20,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 40,
        "margin": 8,
        "marginHorizontal": 40,
        "paddingLeft": 20,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "flexDirection": "row",
        "fontSize": 14,
        "justifyContent": "space-between",
        "marginBottom": 15,
        "marginHorizontal": 30,
        "marginTop": 14,
        "paddingHorizontal": 8,
        "paddingVertical": 4,
        "textDecorationColor": "#7A7A7A",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7A7A7A",
        }
      }
    >
      Are you a Player or Coach?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "marginTop": -20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5BF00",
          "borderColor": "#7a7a7a",
          "borderRadius": 20,
          "borderWidth": 1,
          "fontSize": 50,
          "marginHorizontal": 10,
          "marginTop": 30,
          "opacity": 1,
          "paddingVertical": 10,
          "width": 100,
        }
      }
    >
      <Text
        testID="ChoosePlayerButton1"
      >
        Player
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E5E5E5",
          "borderColor": "#7a7a7a",
          "borderRadius": 20,
          "borderWidth": 1,
          "fontSize": 50,
          "marginHorizontal": 10,
          "marginTop": 30,
          "opacity": 1,
          "paddingVertical": 10,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#7A7A7A",
          }
        }
        testID="ChooseCoachButton1"
      >
        Coach
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#F5BF00",
        "borderColor": "#000000",
        "borderRadius": 20,
        "borderWidth": 1,
        "fontSize": 50,
        "marginTop": 30,
        "opacity": 0.5,
        "paddingHorizontal": 12,
        "paddingVertical": 10,
        "width": 120,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7A7A7A",
        }
      }
      testID="SignUpButton"
    >
      Sign Up
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#E5E5E5",
        "borderColor": "#000000",
        "borderRadius": 20,
        "borderWidth": 1,
        "fontSize": 50,
        "marginTop": 10,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 10,
        "width": 100,
      }
    }
  >
    <Text
      testID="NavigateToLogIn"
    >
      Log in
    </Text>
  </View>
</RCTSafeAreaView>
`;
